### ISSUE TITLE
missing options in Dreambooth training

### ISSUE THREAD

**Issue by b-fission** [bmaltais/kohya_ss#2542]
Two things on Dreambooth training for SDXL
- Missing checkbox for "No half VAE" leading to NaN errors
- TE1/TE2 learning rates have no effect. The training script indicates it was only training the Unet in SDXL dreambooth. I think the gui isn't passing the arg `--train_text_encoder`

**Comment by b-fission** [comment]
It seems the entire "SDXL Specific Parameters" group is missing in the master branch (and release) versions. I even searched the source code for "fused backward pass" and got nothing.

When I switched to dev, those options do show up.

**Comment by bmaltais** [comment]
There are many fixes in dev. I am waiting for Kohya to merge dev to main to promote the dev branch to master. In the mean time you can use the dev branch as it should be more stable than master at this point.

**Comment by rafstahelin** [comment]
@bmaltais I am on dev, and also dont see option to checkbox SDXL? So it seems like there can be no training in SDXL at all for the moment. Basically locked out of training. Unless I am missing something

**Comment by bmaltais** [comment]
It will only show up if you select a custom model instead of the default
huggingface ones.

**Comment by rafstahelin** [comment]
Ok. That might be it, though I loaded my base model from /workspace on runpod
Will check again thx

**Comment by b-fission** [comment]
> @b-fission Does TE1 and TE2 learning rates only apply to Dreambooth? Wanted to try it in Lora module
Yes, TE1 and TE2 are for Dreambooth. Lora training only handles a single TE.

**Comment by rafstahelin** [comment]
I will try it 
Would it be something like 
Train_TE1 = true etc..?

**Comment by rafstahelin** [comment]
I was thinking of turning Te2 off 

By the way for subjects, what worked better for you, dreambooth or Lora implementation?

**Comment by b-fission** [comment]
I think Dreambooth tends to reach the intended results I want with less tweaking compared to Lora. 

Afterwards, I can use the Extract Lora utility to generate a Lora from the generated Dreambooth checkpoint, so I can have the flexibility of Lora and save disk space.

**Comment by rafstahelin** [comment]
I've used both but testing some other optimisers like prodigy and Adafactor besides adamW8bit
Which one do you prefer?

**Comment by b-fission** [comment]
For me, AdamW8bit has worked well enough that I haven't put much time into trying other optimizers.

I've played with Prodigy a few times with Loras and got 'interesting' results, but it sometimes learned too fast and overtrained; probably needs extra setup on my end.

Haven't tried the other ones though. From what little testing I did, Lion8bit and Adafactor are interesting for having lower vram requirements, which makes them viable for training SDXL on mid-range hardware.

**Comment by rafstahelin** [comment]
thanks for the insights, similar to my understanding. I'll ditch adafactor. Interested mainly in adamw8bit vs prodigy. and testing TE1/TE2. Or TrainTE for some % steps. Though not sure that works still in dreambooth

**Comment by b-fission** [comment]
The option for "Stop TE (% of total steps)" only seems to be relevant in Dreambooth for SD 1.x or 2.0, but not SDXL.

**Comment by rafstahelin** [comment]
Unfortunate
